File: software/examples/vbnet/ExampleTouch.vb
Imports System
Imports Tinkerforge

Module ExampleTouch
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your LCD 128x64 Bricklet

    ' Callback subroutine for touch position callback
    Sub TouchPositionCB(ByVal sender As BrickletLCD128x64, ByVal pressure As Integer, _
                        ByVal x As Integer, ByVal y As Integer, ByVal age As Long)
        Console.WriteLine("Pressure: " + pressure.ToString())
        Console.WriteLine("X: " + x.ToString())
        Console.WriteLine("Y: " + y.ToString())
        Console.WriteLine("Age: " + age.ToString())
        Console.WriteLine("")
    End Sub

    ' Callback subroutine for touch gesture callback
    Sub TouchGestureCB(ByVal sender As BrickletLCD128x64, ByVal gesture As Byte, _
                       ByVal duration As Long, ByVal pressureMax As Integer, _
                       ByVal xStart As Integer, ByVal xEnd As Integer, _
                       ByVal yStart As Integer, ByVal yEnd As Integer, _
                       ByVal age As Long)
        Console.WriteLine("Gesture: " + gesture.ToString())
        Console.WriteLine("Duration: " + duration.ToString())
        Console.WriteLine("Pressure Max: " + pressureMax.ToString())
        Console.WriteLine("X Start: " + xStart.ToString())
        Console.WriteLine("X End: " + xEnd.ToString())
        Console.WriteLine("Y Start: " + yStart.ToString())
        Console.WriteLine("Y End: " + yEnd.ToString())
        Console.WriteLine("Age: " + age.ToString())
        Console.WriteLine("")
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim lcd As New BrickletLCD128x64(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Register touch position callback to subroutine TouchPositionCB
        AddHandler lcd.TouchPositionCallback, AddressOf TouchPositionCB

        ' Register touch gesture callback to subroutine TouchGestureCB
        AddHandler lcd.TouchGestureCallback, AddressOf TouchGestureCB

        ' Set period for touch position callback to 0.1s (100ms)
        lcd.SetTouchPositionCallbackConfiguration(100, True)

        ' Set period for touch gesture callback to 0.1s (100ms)
        lcd.SetTouchGestureCallbackConfiguration(100, True)

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/java/ExampleTouch.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletLCD128x64;

public class ExampleTouch {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your LCD 128x64 Bricklet
	private static final String UID = "XYZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletLCD128x64 lcd = new BrickletLCD128x64(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Add touch position listener
		lcd.addTouchPositionListener(new BrickletLCD128x64.TouchPositionListener() {
			public void touchPosition(int pressure, int x, int y, long age) {
				System.out.println("Pressure: " + pressure);
				System.out.println("X: " + x);
				System.out.println("Y: " + y);
				System.out.println("Age: " + age);
				System.out.println("");
			}
		});

		// Add touch gesture listener
		lcd.addTouchGestureListener(new BrickletLCD128x64.TouchGestureListener() {
			public void touchGesture(int gesture, long duration, int pressureMax,
			                         int xStart, int xEnd, int yStart, int yEnd,
			                         long age) {
				System.out.println("Gesture: " + gesture);
				System.out.println("Duration: " + duration);
				System.out.println("Pressure Max: " + pressureMax);
				System.out.println("X Start: " + xStart);
				System.out.println("X End: " + xEnd);
				System.out.println("Y Start: " + yStart);
				System.out.println("Y End: " + yEnd);
				System.out.println("Age: " + age);
				System.out.println("");
			}
		});

		// Set period for touch position callback to 0.1s (100ms)
		lcd.setTouchPositionCallbackConfiguration(100, true);

		// Set period for touch gesture callback to 0.1s (100ms)
		lcd.setTouchGestureCallbackConfiguration(100, true);

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}
